### PR TITLE
Update criteria for .NET Foundation membership

### PR DESCRIPTION
**Pull request by BillWagner**
Prior to this year, the criteria was focused on contributions to .NET open source projects. That's only one possible way to participate.

This PR:

- clarifies what we look for when an open source maintainer or contributor requests to join.
- Adds criteria for those prospective members who identify more closely with other membership activities.

### CODE REVIEW

File: meetings/membership-criteria.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Membership criteria
+
+When you apply for membership, we review your application based on the [membership participation types](membership-participation-types.md) defined. We ask for some basic information about you. 
+
+## Basic information
+
+First, we validate the personal information:
+
+- Reasonable first name, last name.
```
Comment (jguadagno) [inline meetings/membership-criteria.md:9]: Reasonable might be questioned. Instead

First Name and Last Name are reviewed to eliminate potential 'bots'.

File: meetings/membership-criteria.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
```
Comment (jguadagno) [inline meetings/membership-criteria.md:10]: valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
should be
A valid email address: *We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.*

File: meetings/membership-criteria.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+- Responsibility for tasks related to the meetup or conference.
+- Building a community among related groups or conferences. This could be sharing speakers, creating partnered events, or other activities that grow the community.
+
+## Educator
+
+See [Educator particpation type](membership-participation-types.md#the-educator).
+
+We look to the educator's creation and delivery of curriculum in any of the following ways:
+
+- Teaching bootcamp courses, college or high school level courses in software development, where the .NET platform is a signification part of the curriculum.
```
Comment (jguadagno) [inline meetings/membership-criteria.md:44]: Teaching. Whether that be Bootcamp courses, college, or high school level courses in software development where the .NET platform is a signification part of the curriculum.

File: meetings/membership-criteria.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+- Creating .NET based curriculum for any of these area.
+
+## Knowledge sharers
+
+See [Sharer of knowledge particpation type](membership-participation-types.md#the-sharer-of-knowledge).
+
+We look at the content they create and deliver to spread knowledge. These include:
```
Comment (jguadagno) [inline meetings/membership-criteria.md:51]: The content that the applicant either creates and/or delivers to spread knowledge. These include, but are not limited to:

File: meetings/membership-criteria.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+- books
+- conference talks.
```
Comment (jguadagno) [inline meetings/membership-criteria.md:54]: No need for period

File: meetings/membership-criteria.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- twitch streams
+- blogs.
```
Comment (jguadagno) [inline meetings/membership-criteria.md:56]: No need for period

File: meetings/membership-criteria.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+- online courseware.
```
Comment (jguadagno) [inline meetings/membership-criteria.md:57]: No need for period

File: meetings/membership-criteria.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+- stack overflow answers.
```
Comment (jguadagno) [inline meetings/membership-criteria.md:58]: No need for period

**Review by jguadagno — CHANGES_REQUESTED** [review]
I made a few suggestions

File: meetings/membership-criteria.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+- Reasonable first name, last name.
+- valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
+- valid GitHub ID. We validate the profile appears to be human. There should be some activity. The user name must match any contribution links provided. GitHub activity mentioned in the contribution statement should match this ID.
+
+## Open Source Maintainer, Open Source Contributor
+
+See [Open source maintainer](membership-participation-types.md#the-open-source-maintainer) and the [Open source contributor](membership-participation-types.md#the-open-source-contributor).
+
+We look at the contribution links. All activity must conform to the .NET Foundation CoC. Activities examined include:
```
[the remark this answers is not in the document]
Comment (wordshaker) [reply meetings/membership-criteria.md:17]: Would be good to link it as well

**Review by CalvinAllen — APPROVED** [review]
Approved, though I agree with Joe's change requests

**Comment by ovebastiansen** [comment]
Together with Joe's comments this looks good to me

**Comment by CalvinAllen** [comment]
Went ahead and made those changes to save @BillWagner some work :)

**Review by wordshaker — APPROVED** [review]

File: meetings/membership-criteria.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+## Event organizer
+
+See [Event organizer particpation type](membership-participation-types.md#the-event-organizer).
+
+We look at the event or user group activities. Examples include:
+
+- Events listed have a CoC that reflects the .NET Foundation's community guidelines.
+- Consistent track record holding events (virtual or in person).
+- Meetup is listed with the .NET Foundation
```
Comment (BillWagner) [inline meetings/membership-criteria.md:34]: Clarify that this bullet only applies to meetup groups. (Conference organizers need not use meetup)

File: meetings/membership-criteria.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+A person need not participate in all these areas, but should be active in some of them.
+
+## Event organizer
+
+See [Event organizer particpation type](membership-participation-types.md#the-event-organizer).
+
+We look at the event or user group activities. Examples include:
+
+- Events listed have a CoC that reflects the .NET Foundation's community guidelines.
```
Comment (BillWagner) [inline meetings/membership-criteria.md:32]: Add link

Comment (wordshaker) [inline meetings/membership-criteria.md:32]: Add link to the community guidelines?

File: meetings/membership-criteria.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+- Responsibility for tasks related to the meetup or conference.
+- Building a community among related groups or conferences. This could be sharing speakers, creating partnered events, or other activities that grow the community.
+
+## Educator
+
+See [Educator particpation type](membership-participation-types.md#the-educator).
+
+We look to the educator's creation and delivery of curriculum in any of the following ways:
+
+- Teaching. Whether that be bootcamp courses, college or high school level courses in software development where the .NET platform is a signification part of the curriculum.
```
Comment (BillWagner) [inline meetings/membership-criteria.md:44]: Add online platforms (MS Learn, Pluralsight)

File: meetings/membership-criteria.md (hunk 10 above)
Comment (wordshaker) [inline meetings/membership-criteria.md:34]: @ovebastiansen pointed out the meetup being listed with the .NET Foundation is restrictive

File: meetings/membership-criteria.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+See [Sharer of knowledge particpation type](membership-participation-types.md#the-sharer-of-knowledge).
+
+The content that the applicant creates and/or delivers to spread knowledge. These include, but are not limited to:
+
+- books
+- conference talks
+- twitch streams
+- blogs
+- online courseware
+- stack overflow answers
```
Comment (BillWagner) [inline meetings/membership-criteria.md:58]: Point out that these (except maybe books) should show a track record of answers over time)

File: meetings/membership-criteria.md [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+See [Educator particpation type](membership-participation-types.md#the-educator).
+
+We look to the educator's creation and delivery of curriculum in any of the following ways:
+
+- Teaching. Whether that be bootcamp courses, college or high school level courses in software development where the .NET platform is a signification part of the curriculum.
+- Creating .NET based curriculum for any of these area.
+
+## Knowledge sharers
+
+See [Sharer of knowledge particpation type](membership-participation-types.md#the-sharer-of-knowledge).
```
Comment (wordshaker) [inline meetings/membership-criteria.md:49]: particpation

File: meetings/membership-criteria.md [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+- books
+- conference talks
+- twitch streams
+- blogs
+- online courseware
+- stack overflow answers
+
+## Students
+
+See [Student particpation type](membership-participation-types.md#the-student).
```
Comment (wordshaker) [inline meetings/membership-criteria.md:62]: particpation

File: meetings/membership-criteria.md [hunk 16]
```diff
@@ -0,0 +1,64 @@
+# Membership criteria
+
+When you apply for membership, we review your application based on the [membership participation types](membership-participation-types.md) defined. We ask for some basic information about you.
```
Comment (BillWagner) [inline meetings/membership-criteria.md:3]: Fix these links (different folder), etc)

File: meetings/membership-criteria.md [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+
+When you apply for membership, we review your application based on the [membership participation types](membership-participation-types.md) defined. We ask for some basic information about you. 
+
+## Basic information
+
+First, we validate the personal information:
+
+- First and Last name are reviewed to eliminate potential 'bots'.
+- A valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
+- A valid GitHub ID: We validate the profile appears to be human. There should be some activity. The user name must match any contribution links provided. GitHub activity mentioned in the contribution statement should match this ID.
```
Comment (BillWagner) [inline meetings/membership-criteria.md:11]: This requirement can be removed. With the new portal, we don't need a GitHub id to communicate with members.

File: meetings/membership-criteria.md [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+
+## Open Source Maintainer, Open Source Contributor
```
Comment (BillWagner) [inline meetings/membership-criteria.md:13]: Question for the project committee. Could projects be hosted on another open platform? e.g. GitLab, BitBucket.

File: meetings/membership-criteria.md [hunk 19]
```diff
@@ -0,0 +1,64 @@
+# Membership criteria
+
+When you apply for membership, we review your application based on the [membership participation types](membership-participation-types.md) defined. We ask for some basic information about you. 
+
+## Basic information
```
Comment (BillWagner) [inline meetings/membership-criteria.md:5]: In this section, make it clear that applicants can (but are not required) to fill in activities in multiple sections.

File: meetings/membership-criteria.md (hunk 16 above)
Comment (WhiteBlackGoose) [inline meetings/membership-criteria.md:3]:
```suggestion
When you apply for membership, we review your application based on the [membership participation types](../membership-participation-types.md) defined. We ask for some basic information about you. 
```

File: meetings/membership-criteria.md [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+- A valid email address: We only validate that it is a valid email address format. Later, when an invitation is sent, that verifies that it's a real address.
+- A valid GitHub ID: We validate the profile appears to be human. There should be some activity. The user name must match any contribution links provided. GitHub activity mentioned in the contribution statement should match this ID.
+
+## Open Source Maintainer, Open Source Contributor
+
+See [Open source maintainer](membership-participation-types.md#the-open-source-maintainer) and the [Open source contributor](membership-participation-types.md#the-open-source-contributor).
+
+We look at the contribution links. All activity must conform to the .NET Foundation [Code of Conduct (CoC)](https://dotnetfoundation.org/about/code-of-conduct). Activities examined include:
+
+- PRs created. We evaluate the participation in the community. Does the PR match the project's guidelines? Does the author engage constructively with reviewers?
```
Comment (WhiteBlackGoose) [inline meetings/membership-criteria.md:19]:
```suggestion
- Pull Requests (PRs) created. We evaluate the participation in the community. Does the PR match the project's guidelines? Does the author engage constructively with reviewers?
```

File: meetings/membership-criteria.md [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+- PR review, and helping new contributors. Are comments in line with the spirit of the CoC? Does the reviewer provide necessary assistance?
+- Issues created. These should be well-defined issues. We don't evaluate issue quality based on the project maintainer's decision to accept or reject the idea. We do evaluate issues based on the clarity and completeness.
+- Discussion forums. Does the person participate in discussions in a constructive manner?
+
+A person need not participate in all these areas, but should be active in some of them.
+
+## Event organizer
+
+See [Event organizer particpation type](membership-participation-types.md#the-event-organizer).
```
Comment (WhiteBlackGoose) [inline meetings/membership-criteria.md:28]:
```suggestion
See [Event organizer participation type](membership-participation-types.md#the-event-organizer).
```

File: meetings/membership-criteria.md [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+
+- Events listed have a CoC that reflects the .NET Foundation's community guidelines.
+- Consistent track record holding events (virtual or in person).
+- Meetup is listed with the .NET Foundation
+- Responsibility for tasks related to the meetup or conference.
+- Building a community among related groups or conferences. This could be sharing speakers, creating partnered events, or other activities that grow the community.
+
+## Educator
+
+See [Educator particpation type](membership-participation-types.md#the-educator).
```
Comment (WhiteBlackGoose) [inline meetings/membership-criteria.md:40]:
```suggestion
See [Educator participation type](membership-participation-types.md#the-educator).
```

File: meetings/membership-criteria.md (hunk 14 above)
Comment (WhiteBlackGoose) [inline meetings/membership-criteria.md:49]:
```suggestion
See [Sharer of knowledge participation type](membership-participation-types.md#the-sharer-of-knowledge).
```

File: meetings/membership-criteria.md (hunk 11 above)
Comment (WhiteBlackGoose) [inline meetings/membership-criteria.md:32]:
```suggestion
- Events listed have a CoC that reflects the .NET Foundation's [community guidelines](https://dotnetfoundation.org/about/code-of-conduct).
```

File: meetings/membership-criteria.md (hunk 15 above)
Comment (WhiteBlackGoose) [inline meetings/membership-criteria.md:62]:
```suggestion
See [Student participation type](membership-participation-types.md#the-student).
```

**Review by WhiteBlackGoose — COMMENTED** [review]
You can add a recommendation to fix a line with suggestion (when reviewing a line, there's a button "add a suggestion"), if it makes it easier

**Comment by BillWagner** [comment]
> You can add a recommendation to fix a line with suggestion (when reviewing a line, there's a button "add a suggestion"), if it makes it easier

We do usually do that. We were discussing live in our meeting yesterday and adding quick notes for our editing cycle. Thanks for adding all the suggestions. It save quite a bit of time.

**Comment by BillWagner** [comment]
Thanks @jguadagno @CalvinAllen @wordshaker I've addressed all the comments from our meeting. Do you want to give it a last read?  I've also pinged the project committee to find out if we look at other OSS hosting options today.

File: meetings/membership-criteria.md (hunk 18 above)
Comment (tannergooding) [reply meetings/membership-criteria.md]: Shaun should comment on this one. I don't think it's come up yet, however.

**Review by jguadagno — APPROVED** [review]

**Review by CalvinAllen — APPROVED** [review]

**Comment by CalvinAllen** [comment]
Ship it! 🚀

**Pull request merged**